docs/source/frameworks/transformers.rst

```diff
@@ -1,38 +1,312 @@
+    from transformers import pipeline
+
+    class PretrainedModelRunnable(bentoml.Runnable):
+        SUPPORTED_RESOURCES = ("cpu",)
+        SUPPORTS_CPU_MULTI_THREADING = True
+
+        def __init__(self):
+            self.unmasker = pipeline(task="fill-mask", model="distilbert-base-uncased")
+
+        @bentoml.Runnable.method(batchable=False)
+        def __call__(self, input_text):
+            return self.unmasker(input_text)
+
+    runner = bentoml.Runner(PretrainedModelRunnable, name="pretrained_unmasker")
+
+    svc = bentoml.Service('pretrained_unmasker_service', runners=[runner])
+
+    @svc.api(input=Text(), output=JSON())
+    def unmask(input_series: str) -> list:
+        return runner.run(input_series)
+
+Custom Pipelines
+----------------
+
+Transformers custom pipelines allow users to define their own pre and post-process logic and customize how input data is forwarded to 
+the model for inference.
+
+.. seealso::
+
+    `How to add a pipeline <https://huggingface.co/docs/transformers/main/en/add_new_pipeline>`_ from Hugging Face to learn more.
+
+.. code-block:: python
+    :caption: train.py
+    
+    from transformers import Pipeline
+
+    class MyClassificationPipeline(Pipeline):
+        def _sanitize_parameters(self, **kwargs):
+            preprocess_kwargs = {}
+            if "maybe_arg" in kwargs:
+                preprocess_kwargs["maybe_arg"] = kwargs["maybe_arg"]
+            return preprocess_kwargs, {}, {}
+
+        def preprocess(self, text, maybe_arg=2):
+            input_ids = self.tokenizer(text, return_tensors="pt")
+            return input_ids
+
+        def _forward(self, model_inputs):
+            outputs = self.model(**model_inputs)
+            return outputs
+
+        def postprocess(self, model_outputs):
+            return model_outputs["logits"].softmax(-1).numpy()
+
+Saving a Custom Pipeline
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+A custom pipeline first needs to be added to the Transformers supported tasks, :code:`SUPPORTED_TASKS` before it can be created with 
+the Transformers :code:`pipeline` API.
+
+.. code-block:: python
+    :caption: train.py
+    
+    from transformers import pipeline
+    from transformers import AutoTokenizer
+    from transformers import AutoModelForSequenceClassification
+    from transformers.pipelines import SUPPORTED_TASKS
+
+    TASK_NAME = "my-classification-task"
+    TASK_DEFINITION = {
+        "impl": MyClassificationPipeline,
+        "tf": (),
+        "pt": (AutoModelForSequenceClassification,),
+        "default": {},
+        "type": "text",
+    }
+    SUPPORTED_TASKS[TASK_NAME] = TASK_DEFINITION
+
+    classifier = pipeline(
+        task=TASK_NAME,
+        model=AutoModelForSequenceClassification.from_pretrained(
+            "distilbert-base-uncased-finetuned-sst-2-english"
+        ),
+        tokenizer=AutoTokenizer.from_pretrained(
+            "distilbert-base-uncased-finetuned-sst-2-english"
+        ),
+    )
```

Comment:
cc @aarnphm: backwards compatibility for older transformers 4.x when supporting custom pipelines when the new pipeline registy is out.